src/bio.lisp

```diff
@@ -203,9 +203,8 @@
             do
                (setf char (read-byte *socket*)
                      exit (= char 10))
-               (unless exit
-                 (setf (cffi:mem-ref buf :unsigned-char i) char)
-                 (incf i))))
+               (setf (cffi:mem-ref buf :unsigned-char i) char)
+               (incf i)))
```

Comment:
You decided to count the newline?

The docs may be interpretted that way, but 
OpenSSL implementanion uses `strlen` on the buffer, thus not counting newline.
https://github.com/openssl/openssl/blob/master/crypto/bio/bss_file.c#L347

Reply:
I did, based on:
https://www.openssl.org/docs/man1.1.1/man3/BIO_gets.html

The returned string is always NUL-terminated and the '\n' is preserved if present in the input data.
...
The trailing NUL is not included in the length returned by BIO_gets().
...and section HISTORY:
BIO_gets() on 1.1.0 and older when called on BIO_fd() based BIO does not keep the '\n' at the end of the line in the buffer._

Reply:
Right, So the semantics is based on `fgets`, which, unlike `gets`, inclues the new newline character into the `buf`.